tapestry.js

```diff
@@ -105,25 +105,36 @@ jQuery.get(apiUrl + "/tapestries/" + tapestryWpPostId, function(result){
     // 2. SIZE AND SCALE THE TAPESTRY AND SVG TO FIT WELL
     //---------------------------------------------------
 
+
+    // establish initial positioning before the startforce()
+    // this so 'getNodeDimensions' is operational at this stage
+    for (var index in dataset.nodes) {
+        if (dataset.nodes[index].x == null) {
+            dataset.nodes[index].x = getBrowserWidth()*0.9;
+        }
+        if (dataset.nodes[index].y == null) {
+            dataset.nodes[index].y = getBrowserHeight()*0.9;
+        }
+    }
+
     function updateTapestrySize() {
 
         var nodeDimensions = getNodesDimensions(dataset);
     
         // Transpose the tapestry so it's longest side is aligned with the longest side of the browser
         // For example, vertically long tapestries should be transposed so they are horizontally long on desktop,
         // but kept the same way on mobile phones where the browser is vertically longer
-        var tapestryAspectRatio = nodeDimensions['x'] / nodeDimensions['y'];
+        var tapestryAspectRatio = nodeDimensions.x / nodeDimensions.y;
         var windowAspectRatio = getAspectRatio();
-        if (tapestryAspectRatio > 1 && windowAspectRatio < 1 || tapestryAspectRatio < 1 && windowAspectRatio > 1) {
-            transposeNodes();
+        if (tapestryAspectRatio >= 1 && windowAspectRatio <= 1 || tapestryAspectRatio <= 1 && windowAspectRatio >= 1) {
+             transposeNodes();
         }
         
         // Update svg dimensions to the new dimensions of the browser
         updateSvgDimensions(TAPESTRY_CONTAINER_ID);
     }
 
-    // do it now
-    updateTapestrySize();
+    //updateTapestrySize();
```

Comment:
If this is not needed, please remove it and also update the comment for the next line since it says "also ..."

Reply:
this is from before I started working on tapestry (`commit 8ea3653fa84773ec06100e60224f070e3b9a2a35`), but it seems to work anyway. I'll delete it but I'm not sure what purpose it had originally?